netket/sampler/rules/custom_numpy.py

```diff
@@ -76,17 +72,9 @@ def transition(rule, sampler, machine, parameters, state, rng, σ):
         # numba does not support jitting np.random number generators
         # so we have to generate the random numbers outside the jit
         # block
-        rnd_uniform = rng.uniform(0.0, 1.0, size=σ.shape[0])
-
-        _pick_random_and_init(
-            σ.shape[0],
-            rule_state.weight_cumsum,
-            rnd_uniform=rnd_uniform,
-            out=rule_state.rand_op_n,
-        )
-
+        filters = rng.integers(0, len(rule.weight_list), size=σ.shape[0])
```

Comment:
I think this is wrong...
shouldn't you sample the integers with a given probability distribution (aka `rule_state.weight_cumsum`)

Reply:
btw, since tests are passing, if this is wrong, it means that this is not tested..

Aka: a test would be great.

Reply:
yes this looks wrong

Reply:
we should test this with a hamiltonian with random matrix elements

Reply:
I have changed the random number generator, and I have added a meaningful test for this.